novelai-api/api.go

```diff
@@ -323,17 +349,32 @@ func (api NovelAiAPI) GenerateWithParams(content *string, params NaiGeneratePara
 	msg := NewGenerateMsg(encodedBytes64)
 	msg.Parameters = params
 	apiResp := naiApiGenerate(&api.keys, msg, api.backend)
-	if binTokens, err := base64.StdEncoding.DecodeString(apiResp.Output); err != nil {
-		log.Println("ERROR:", err)
-		resp.Error = err
-	} else {
-		tokens := gpt_bpe.TokensFromBin(&binTokens)
-		if params.TrimResponses != nil && *params.TrimResponses == true {
-			tokens, err = api.encoder.TrimIncompleteSentence(tokens)
+	if params.NextWord == false {
+		if binTokens, err := base64.StdEncoding.DecodeString(apiResp.Output); err != nil {
+			log.Println("ERROR:", err)
+			resp.Error = err
+		} else {
+			tokens := gpt_bpe.TokensFromBin(&binTokens)
+			if params.TrimResponses != nil && *params.TrimResponses == true {
+				tokens, err = api.encoder.TrimIncompleteSentence(tokens)
+			}
+			resp.EncodedResponse = apiResp.Output
+			resp.Response = api.encoder.Decode(tokens)
+		}
+	}
+
+	if params.NextWord == true {
+		err := json.Unmarshal([]byte(apiResp.Output), &val)
+		if err != nil {
+			log.Printf("API: Error unmarshaling JSON NextWord response: %s %s", err, apiResp.Output)
+			fmt.Scanln()
+			os.Exit(1)
 		}
-		resp.EncodedResponse = apiResp.Output
-		resp.Response = api.encoder.Decode(tokens)
+
+		str := fmt.Sprintf("%v", val)
+		fmt.Println("\033[38;5;240m" + str)
```

Comment:
We should probably extend the `resp` structure, rather than use `fmt.Println` as a side effect function.  I can fix this.